pkg/api/v1alpha1/cloudstackdatacenterconfig_webhook.go

```diff
@@ -104,21 +104,21 @@ func validateImmutableFieldsCloudStackCluster(new, old *CloudStackDatacenterConf
 	if zonesMutated {
 		allErrs = append(
 			allErrs,
-			field.Invalid(field.NewPath("spec", "zone"), new.Spec.Zones, "field is immutable"),
+			field.Forbidden(field.NewPath("spec", "zone"), "field is immutable"),
```

Comment:
Should this be the following? But check the spec fields in the api package
```suggestion
			field.Forbidden(field.NewPath("spec", "zones"), "field is immutable"),
```